### ISSUE TITLE
Make `EVM.db` not optional

### ISSUE THREAD

**Issue by DaniPopes** [bluealloy/revm#697]
Given that basically every `EVM` method requires that the `db` field is `Some`, we should remove the option entirely.

It would look something like this:

```diff
--- a/crates/revm/src/evm.rs
+++ b/crates/revm/src/evm.rs
@@ -41,16 +41,16 @@ use revm_precompile::Precompiles;
 #[derive(Clone)]
 pub struct EVM<DB> {
     pub env: Env,
-    pub db: Option<DB>,
+    pub db: DB,
 }
 
-pub fn new<DB>() -> EVM<DB> {
-    EVM::new()
+pub fn new<DB>(db: DB) -> EVM<DB> {
+    EVM::new(db)
 }
 
-impl<DB> Default for EVM<DB> {
+impl<DB: Default> Default for EVM<DB> {
     fn default() -> Self {
-        Self::new()
+        Self::new(Default::default())
     }
 }


```

**Comment by alessandromazza98** [comment]
I could take this one. It's my first issue for Revm!

**Comment by rakita** [comment]
> I could take this one. It's my first issue for Revm!

go for it

**Comment by luffyforever** [comment]
> Given that basically every `EVM` method requires that the `db` field is `Some`, we should remove the option entirely.
> 
> It would look something like this:
> 
> ```diff
> --- a/crates/revm/src/evm.rs
> +++ b/crates/revm/src/evm.rs
> @@ -41,16 +41,16 @@ use revm_precompile::Precompiles;
>  #[derive(Clone)]
>  pub struct EVM<DB> {
>      pub env: Env,
> -    pub db: Option<DB>,
> +    pub db: DB,
>  }
>  
> -pub fn new<DB>() -> EVM<DB> {
> -    EVM::new()
> +pub fn new<DB>(db: DB) -> EVM<DB> {
> +    EVM::new(db)
>  }
>  
> -impl<DB> Default for EVM<DB> {
> +impl<DB: Default> Default for EVM<DB> {
>      fn default() -> Self {
> -        Self::new()
> +        Self::new(Default::default())
>      }
>  }
> ```

What is the recommended method for initializing the database after this change? Any guidance or suggestions would be appreciated.

**Comment by rakita** [comment]
Closed with #888
